conda/recipes/ucxx/recipe.yaml

```
@@ -0,0 +1,244 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/prefix-dev/recipe-format/main/schema.json
+# Copyright (c) 2019-2025, NVIDIA CORPORATION.
+
+context:
+  version: ${{ env.get("RAPIDS_PACKAGE_VERSION") }}
+  cuda_version: ${{ (env.get("RAPIDS_CUDA_VERSION") | split("."))[:2] | join(".") }}
+  cuda_major: ${{ (env.get("RAPIDS_CUDA_VERSION") | split("."))[0] }}
+  date_string: ${{ env.get("RAPIDS_DATE_STRING") }}
+  py_version: ${{ env.get("RAPIDS_PY_VERSION") }}
+  head_rev: ${{ git.head_rev(".")[:8] }}
+
+recipe:
+  name: libucxx-split
+
+cache:
+  source:
+    path: ../../..
+
+  requirements:
+    build:
+      - cmake ${{ cmake_version }}
+      - ninja
+      - ${{ compiler("c") }}
+      - ${{ compiler("cxx") }}
+      - ${{ compiler("cuda") }}
+      - cuda-version =${{ cuda_version }}
+      - ${{ stdlib("c") }}
+    host:
+      - cuda-version =${{ cuda_version }}
+      - if: cuda_major == "11"
+        then: cuda-cudart-dev =${{ cuda_version }}
```

This fixed this locally for me with cuda 11 -- not sure if `cuda-cudart-dev` is constrained in some weird way version-wise, or if `rattler-build` is doing something unexpected re: version solving

---

(^everything Vyas said, and...)

CUDA 12 builds should not need the `nvidia` channel for anything. The only CUDA 11 packages we need from the `nvidia` channel are math libraries, so that we can get the C++ headers that don't ship with `conda-forge::cudatoolkit` for CUDA 11. I think that only affects RAFT/cuML/cuGraph/cuVS.

We are currently only including the `nvidia` channel for consistency with CUDA 11, and will drop it from our build commands once CUDA 11 is no longer supported. Perhaps we should even consider dropping `nvidia` now, for the packages that don't need its math library headers.

---

What would we prefer?

- drop `-c nvidia` from the `rapids-rattler-channel-string` and then add it piecemeal to raft, cuml, cugraph, and cuvs?
- wait until we drop cuda 11

---

I think `rapids-rattler-channel-string` should check the value of `RAPIDS_CUDA_VERSION` and only add `--channel nvidia` on CUDA 11. We can implement that as part of the rattler-build migration, no need to wait until we drop CUDA 11.

---

The version of this package should be constrained by `cuda-version`. We should never pin CTK package versions in our recipes (for CUDA 12 only -- [CUDA 11 requires exact pinnings like these](https://github.com/rapidsai/raft/blob/0ba3cc00fe38dc10d757ed63089e00355a738d54/conda/recipes/libraft/conda_build_config.yaml#L22-L48)). CUDA 12 packages should always float to whatever is allowed by `cuda-version`.
```suggestion
        then: cuda-cudart-dev
```